apps/io-profile-async/src/utils/healthcheck.ts

```diff
@@ -0,0 +1,145 @@
+
+// format and cast a problem message with its source
+const formatProblem = <S extends ProblemSource>(
+  source: S,
+  message: string
+): HealthProblem<S> => `${source}|${message}` as HealthProblem<S>;
+
+// utility to format an unknown error to an arry of HealthProblem
+const toHealthProblems = <S extends ProblemSource>(source: S) => (
+  e: unknown
+): ReadonlyArray<HealthProblem<S>> => [
+  formatProblem(source, toError(e).message)
+];
+
+/**
+ * Check application's configuration is correct
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkConfigHealth = (): HealthCheck<"Config", IConfig> =>
+  pipe(
+    getConfig(),
+    TE.fromEither,
+    TE.mapLeft(errors =>
+      // give each problem its own line
+      errors.map(e => formatProblem("Config", readableReport([e])))
+    )
+  );
+
+/**
+ * Check the application can connect to an Azure CosmosDb instances
+ *
+ * @param dbUri uri of the database
+ * @param dbUri connection string for the storage
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkAzureCosmosDbHealth = (
+  dbUri: string,
+  dbKey?: string
+): HealthCheck<"AzureCosmosDB", true> =>
+  pipe(
+    TE.tryCatch(() => {
+      const client = new CosmosClient({
+        endpoint: dbUri,
+        key: dbKey
+      });
+      return client.getDatabaseAccount();
+    }, toHealthProblems("AzureCosmosDB")),
+    TE.map(_ => true)
+  );
+
+/**
+ * Check the application can connect to an Azure Storage
+ *
+ * @param connStr connection string for the storage
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkAzureStorageHealth = (
+  connStr: string
+): HealthCheck<"AzureStorage"> =>
+  pipe(
+    [TableServiceClient.fromConnectionString],
+    // for each, create a task that wraps getServiceProperties
+    RA.map(createService =>
+      TE.tryCatch(
+        () =>
+          new Promise<GetPropertiesResponse>((resolve, reject) =>
+            createService(connStr)
+              .getProperties()
+              .then(
+                result => {
+                  resolve(result);
+                },
+                err => {
+                  reject(err.message.replace(/\n/gim, " ")); // avoid newlines
+                }
+              )
+          ),
+        toHealthProblems("AzureStorage")
+      )
+    ),
+    TE.sequenceSeqArray,
+    TE.map(_ => true)
+  );
+
+/**
+ * Check a url is reachable
+ *
+ * @param url url to connect with
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkUrlHealth = (url: string): HealthCheck<"Url", true> =>
+  pipe(
+    TE.tryCatch(() => fetch(url, { method: "HEAD" }), toHealthProblems("Url")),
+    TE.map(_ => true)
+  );
+
+/**
+ * Execute all the health checks for the application
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkApplicationHealth = (): HealthCheck<ProblemSource, true> =>
```

Comment:
It seems we got two ways for making the healthcheck. Please remove the unused one.